### PR TITLE
feat: reselct field so that GTC is default on limit

### LINKED ISSUE #940: Good 'till Canceled should be the default time in force for limit orders. Not IOC
Good `til canceled is a better default time in force than IOC for limit order.
It is a more common default.
It also works both auction and continuous modes.

However, Good till canceled is applicable for market orders. So IOC should remain the default for Market orders

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #940 

# Description ℹ️ & Technical 👨‍🔧

re-select field so that GTC is default on limit  orders and IOC on market orders.

# Demo 📺

image

### CODE REVIEW

File: libs/deal-ticket/src/components/time-in-force-selector.tsx [hunk 1]
```diff
@@ -44,7 +45,13 @@ export const TimeInForceSelector = ({
             timeInForce === VegaWalletOrderTimeInForce.FOK ||
             timeInForce === VegaWalletOrderTimeInForce.IOC
         );
-
+  useEffect(() => {
```
Comment (MadalinaRaicu) [inline libs/deal-ticket/src/components/time-in-force-selector.tsx:48]: TIF field is already selected on Market tab, so whenever order type updates we need to reselect TIF field.

File: .gitignore [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 # IDE - VSCode
 .vscode/*
-!.vscode/settings.json
+.vscode/settings.json
```
Comment (MadalinaRaicu) [inline .gitignore:24]: settings.json was not ignored.

**Review by sam-keen — APPROVED** [review]

File: libs/deal-ticket/src/components/time-in-force-selector.tsx (hunk 1 above)
Comment (mattrussell36) [inline libs/deal-ticket/src/components/time-in-force-selector.tsx:48]: chore (blocking): I'd prefer not to use useEffect like this? Lets just change the TIF state in the function where order type changes.

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/deal-ticket/src/components/time-in-force-selector.tsx (hunk 1 above)
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/time-in-force-selector.tsx]: updated

**Review by mattrussell36 — APPROVED** [review]

**Comment by AndyWhiteVega** [comment]
Tested: works fine

**Pull request merged**